### PR TITLE
Don't apply the unavailability of an extension to members imported from Clang

### PR DESCRIPTION
**Pull request by xymus**

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: lib/AST/Attr.cpp [hunk 1]
```diff
@@ -1073,8 +1073,11 @@ const AvailableAttr *AvailableAttr::isUnavailable(const Decl *D) {
     return attr;
 
   // If D is an extension member, check if the extension is unavailable.
-  if (auto ext = dyn_cast<ExtensionDecl>(D->getDeclContext()))
-    return AvailableAttr::isUnavailable(ext);
+  //
+  // Skip decls imported from Clang: rdar://problem/53956555
```
Comment (jrose-apple) [inline lib/AST/Attr.cpp]: Can you add more detail here for people outside of Apple?

**Comment by jrose-apple** [comment]
…and a test case.

**Comment by xymus** [comment]
@swift-ci Please smoke test

File: lib/AST/Attr.cpp (hunk 1 above)
Comment (xymus) [reply lib/AST/Attr.cpp]: I've added some details, and the test should help to understand the problem too.

File: lib/AST/Attr.cpp [hunk 2]
```diff
@@ -1073,8 +1073,15 @@ const AvailableAttr *AvailableAttr::isUnavailable(const Decl *D) {
     return attr;
 
   // If D is an extension member, check if the extension is unavailable.
-  if (auto ext = dyn_cast<ExtensionDecl>(D->getDeclContext()))
-    return AvailableAttr::isUnavailable(ext);
+  //
+  // Skip decls imported from Clang, they could be associated to the wrong
+  // extension and inherit undesired unavailability. The ClangImporter
+  // associates Objective-C protocol members to the first category following
+  // the protocol declaration, no matter its availability and the availability
```
Comment (jrose-apple) [inline lib/AST/Attr.cpp]: It's "the first category where the protocol is directly or indirectly adopted". "First category following the protocol declaration" sounds like it's about where the `@protocol` is.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Review by jrose-apple — APPROVED** [review]

**Pull request merged**